src/withAuthUser.js

```diff
@@ -160,6 +167,36 @@ const withAuthUser =
         }
         router.replace(destination)
       }, [router, AuthUser])
+      const redirectToCustom = useCallback(() => {
+        logDebug('Redirecting to user-specified endpoint')
+        const redirectConfig = onRedirect || getConfig().onRedirect
+        if (!redirectConfig) {
+          throw new Error(
+            'The "onRedirect" config setting must be set when using `REDIRECT`.'
+          )
+        }
+
+        const authStateConfig = isAuthed
+          ? onRedirect.whenAuthed
+          : onRedirect.whenUnauthed
+
+        if (!authStateConfig || !authStateConfig.destination) {
+          throw new Error(
+            `The "destination" in the "onRedirect.whenAuthed" and "onRedirect.whenUnauthed" redirect configs must resolve to a non-empty string`
```

Comment:
Nit: We could technically make this only call out the offender here instead of both (with a conditional like you did above)